client/client.go

```diff
@@ -0,0 +1,1662 @@
+}
+
+func (request SdkClientReadRequest) Body(body ClientReadRequest) SdkClientReadRequest {
+	request.body = &body
+	return request
+}
+
+func (request SdkClientReadRequest) Execute() (ClientReadResponse, error) {
+	return request.Client.ReadExecute(request)
+}
+
+func (client *OpenFgaClient) ReadExecute(request SdkClientReadRequest) (ClientReadResponse, error) {
+	data, _, err := client.OpenFgaApi.Read(request.ctx).Body(openfga.ReadRequest{
+		TupleKey: &openfga.TupleKey{
+			User:     request.body.User,
+			Relation: request.body.Relation,
+			Object:   request.body.Object,
+		},
+		PageSize:          getPageSizeFromRequest((*ClientPaginationOptions)(request.options)),
+		ContinuationToken: getContinuationTokenFromRequest((*ClientPaginationOptions)(request.options)),
+	}).Execute()
+
+	return data, err
+}
+
+// / Write
+type SdkClientWriteRequest struct {
+	ctx    _context.Context
+	Client OpenFgaClient
+
+	body    *ClientWriteRequest
+	options *ClientWriteOptions
+}
+
+type ClientWriteRequest struct {
+	Writes  *[]ClientTupleKey
+	Deletes *[]ClientTupleKey
+}
+
+type TransactionOptions struct {
+	// If set to true will disable running in transaction mode (transaction mode means everything is sent in a single transaction to the server)
+	Disable bool `json:"disable,omitempty"`
+	// When transaction mode is disabled, the requests are chunked and sent separately and each chunk is a transaction (default = 1)
+	MaxPerChunk int32 `json:"max_per_chunk,omitempty"`
+	// Number of requests to issue in parallel
+	MaxParallelRequests int32 `json:"max_parallel_requests,omitempty"`
+}
+
+type ClientWriteOptions struct {
+	AuthorizationModelId *string             `json:"authorization_model_id,omitempty"`
+	Transaction          *TransactionOptions `json:"transaction_options,omitempty"`
+}
+
+type ClientWriteStatus string
+
+// List of ClientWriteStatus
+const (
+	SUCCESS ClientWriteStatus = "CLIENT_WRITE_STATUS_SUCCESS"
+	FAILURE ClientWriteStatus = "CLIENT_WRITE_STATUS_FAILURE"
+)
+
+//var allowedTupleOperationEnumValues = []ClientWriteStatus{
+//	SUCCESS,
+//	FAILURE,
+//}
+
+type ClientWriteSingleResponse struct {
+	TupleKey     ClientTupleKey     `json:"tuple_key,omitempty"`
+	Status       ClientWriteStatus  `json:"status,omitempty"`
+	HttpResponse *_nethttp.Response `json:"http_response,omitempty"`
+	Error        error              `json:"error,omitempty"`
+}
+
+func (o ClientWriteSingleResponse) MarshalJSON() ([]byte, error) {
+	toSerialize := map[string]interface{}{}
+	toSerialize["tuple_key"] = o.TupleKey
+	toSerialize["status"] = o.Status
+	if o.HttpResponse != nil {
+		toSerialize["http_response"] = o.HttpResponse
+	}
+	if o.Error != nil {
+		toSerialize["error"] = o.Error
+	}
+	return json.Marshal(toSerialize)
+}
+
+type ClientWriteResponse struct {
+	Writes  []ClientWriteSingleResponse `json:"writes,omitempty"`
+	Deletes []ClientWriteSingleResponse `json:"deletes,omitempty"`
+}
+
+func (o ClientWriteResponse) MarshalJSON() ([]byte, error) {
```

Comment:
this appears to be unused, but if it's not, line 948 should be `toSerialize["deletes"] = o.Deletes`